### ISSUE TITLE
Permanent intrinsic invisibility

### ISSUE THREAD

**Issue by vultur-cadens** [NetHack/NetHack#690]
I believe permanent intrinsic invisibility is too easy to get, and it makes too many items practically useless.

Permanent intrinsic invisibility can be acquired in multiple ways:
* Zap a wand of make invisible at yourself.  Wands of make invisible are fairly common since they are generated as miscellaneous items in monster inventory.  Monsters don't use them up since they only zap themselves once.
* Quaff a blessed potion of invisibility.
* Eat a stalker corpse while (temporarily or extrinsically) invisible.
* Eat a ring of invisibility, sometimes.
* Have it toggled on by a magic trap, since 2bd43869564a9ec5621c8fc1c082cbd66f74817e.

Permanent intrinsic invisibility is unlikely to be lost.  The following are the only ways to lose it that I know of:
* Get hit by a gremlin at night or eat a disenchanter, sometimes.
* Have it toggled off by a magic trap, since 2bd43869564a9ec5621c8fc1c082cbd66f74817e.

Once you are permanently invisible, the following items become mostly useless:
* Wand of make invisible (You only need one charge, so why does it come with multiple charges?  Yeah, you might rarely find it useful against gaze attackers if you don't see invisible, but a blindfold works too.  Invisibility is also useless for pets.)
* Non-cursed potions of invisibility (cursed ones have the use of waking covetous monsters)
* Extrinsic invisibility from invoking the Orb of Detection.  It can't even toggle your intrinsic invisibility *off*.
* Spellbook of invisibility
* Ring of invisibility
* Cloak of invisibility
* Cloak of displacement becomes useless against monsters that don't see invisible.

There are a few mildly negative effects of being invisible:
* Shopping is annoying (put on a mummy wrapping and take it off when you're done).
* Getting monsters to chase you is harder.
* Dealing with Medusa is a little bit harder.
* Conflict doesn't work against monsters that don't see invisible.
* Displacement is mostly useless, but invisibility mostly substitutes for it.
* Locating your character on the screen might be a little bit harder if you don't see invisible.

Maybe the DevTeam originally intended "You made yourself permanently invisible?  Haha, now getting into shops is permanently annoying!  So think carefully before getting permanent invisibility."  But I don't think that's worth having so many items become redundant.

Compare invisibility with another permanent intrinsic that's easy to get: speed.  It is easily gained from a wand of speed, but is inferior to extrinsic and temporary intrinsic speed, so other speed items don't become useless.  Also, over the course of a game, speed has a decent chance of being removed by several monsters (cockatrices, skeletons, shades) and is actually useful for pets, so there's a good chance that players might find a wand of speed useful even after gaining permanent intrinsic speed.  This is not the case for permanent intrinsic invisibility, which is as good as the extrinsic or temporary intrinsic, and is unlikely to be removed.

So I have some suggestions:
* Remove permanent intrinsic invisibility, or make it harder to get.  Every current source of permanent intrinsic invisibility should instead give temporary invisibility.
  * Maybe eating a ring could be the only exception, since the relevant ring is not always edible and it's a relatively difficult way to gain an intrinsic.
  * Mummy wrapping becomes less useful since players can just wait out the invisibility.  Probably still useful for players who have stacked a lot of turns of temporary invisibility and don't want to wait it out.
* Or, if you insist on retaining permanent intrinsic invisibility, make it worse than the extrinsic in some way (other than shops being annoying).
  * Maybe you have a chance of losing intrinsic invisibility each time you attack or get attacked.

I believe @copperwater's xNetHack removed permanent intrinsic invisibility, so maybe they have other thoughts about this.

**Comment by RojjaCebolla** [comment]
we talked about a few ways of fixing this situation with this in splicehack, if it got sorted out upstream that'd be wonderful!

**Comment by copperwater** [comment]
I agree with removing permanent intrinsic invisibility, and also permanent intrinsic see invisible (because that is acquired fairly early on and then for the rest of the game means invisible monsters don't work as a tactic, contributing to the "all combat in the later game feels the same" problem).

For those who don't know, the xNetHack approach is to take all former sources of permanent intrinsic invisibility/see invisible and assign durations of the temporary intrinsic instead. I don't have the numbers at hand but they're all in the hundreds of turns, and see invisible from a throne is thousands. Extrinsic sources are unchanged; it's quite fair to spend a cloak or ring slot on having one of these permanently.

EvilHack *mostly* does this also, but also allows magic traps to toggle both intrinsics; I disagree with that approach because it encourages magic trap camping, which is already incentivized for monster summons, Pw max and charisma increases, and pets, and really doesn't need any more reasons to do it.

I also think the removal of permanent invisibility opens up some interesting avenues for game balance such as "conflict only works on sighted monsters if they can see you wearing conflict". Clearly this isn't that useful of an idea in the game's current state since most people get perma-invisible.

**Comment by entrez** [comment]
> I also think the removal of permanent invisibility opens up some interesting avenues for game balance such as "conflict only works on sighted monsters if they can see you wearing conflict". Clearly this isn't that useful of an idea in the game's current state since most people get perma-invisible.

Agreed this works better with temporary invisibility; as the rule is currently implemented in 3.7, players who zap themselves with invisibility to improve their odds of survival early on are more-or-less permanently neutering their ability to generate conflict (quite possibly without realizing it until much later in the game).  That's not exactly an uninteresting choice (when made as a deliberate choice), but I think it's a lot more fun as a tactical decision you have to make multiple times, under varying circumstances.
